ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -139,134 +132,117 @@ private[kvutils] class ProcessTransactionSubmission(
+              digest,
               RejectionReason.SubmitterCannotActViaParticipant(
-                s"Party '${transactionEntry.submitter}' not hosted by participant $participantId"))
+                s"Party '${digest.submitter}' not hosted by participant $ctx.getParticipantId"))
           )
       case None =>
         reject(
-          recordTime,
+          ctx.getRecordTime,
           buildRejectionLogEntry(
-            transactionEntry,
+            digest,
             RejectionReason.PartyNotKnownOnLedger(
-              s"Submitting party '${transactionEntry.submitter}' not known"))
+              s"Submitting party '${digest.submitter}' not known"))
         )
     }
+  }
 
   /** Validate ledger effective time and the command's time-to-live. */
-  private def validateLedgerTime(
-      recordTime: Timestamp,
-      transactionEntry: TransactionEntry,
-      inputState: DamlStateMap,
-  ): Commit[Unit] = delay {
-    val (_, config) = Common.getCurrentConfiguration(defaultConfig, inputState, logger)
+  private def validateLedgerTime: Step = (ctx, digest) => {
+    val (_, config) = getCurrentConfiguration(defaultConfig, ctx.inputs, logger)
     val timeModel = config.timeModel
-    val givenLedgerTime = transactionEntry.ledgerEffectiveTime.toInstant
+    val givenLedgerTime = digest.ledgerEffectiveTime.toInstant
 
     timeModel
-      .checkTime(ledgerTime = givenLedgerTime, recordTime = recordTime.toInstant)
+      .checkTime(ledgerTime = givenLedgerTime, recordTime = ctx.getRecordTime.toInstant)
       .fold(
         reason =>
           reject(
-            recordTime,
-            buildRejectionLogEntry(transactionEntry, RejectionReason.InvalidLedgerTime(reason))),
-        _ => pass)
+            ctx.getRecordTime,
+            buildRejectionLogEntry(digest, RejectionReason.InvalidLedgerTime(reason))),
+        _ => StepContinue(digest))
   }
 
   /** Validate the submission's conformance to the DAML model */
-  private def validateModelConformance(
-      engine: Engine,
-      recordTime: Timestamp,
-      participantId: ParticipantId,
-      transactionEntry: TransactionEntry,
-      inputState: DamlStateMap,
-  ): Commit[Unit] = delay {
-    // Pull all keys from referenced contracts. We require this for 'fetchByKey' calls
-    // which are not evidenced in the transaction itself and hence the contract key state is
-    // not included in the inputs.
-    lazy val knownKeys: Map[DamlContractKey, Value.AbsoluteContractId] =
-      inputState.collect {
-        case (key, Some(value))
-            if value.hasContractState
-              && value.getContractState.hasContractKey
-              && contractIsActiveAndVisibleToSubmitter(transactionEntry, value.getContractState) =>
-          value.getContractState.getContractKey -> Conversions.stateKeyToContractId(key)
-      }
+  private def validateModelConformance: Step =
+    (ctx, digest) =>
+      metrics.daml.kvutils.committer.transaction.interpretTimer.time(() => {
```

Comment:
Can this be shortened?

Reply:
I find it quite handy because one can see the full metric patch.

Reply:
True, I was thinking about aliasing with a shorter but still meaningful name such as `commitInterpreterTime`, the long name would only be a click away. Disavantage: not one click away on Github so, on second thought, I withdraw it.